Client/Frontend/ContentBlocker/ContentBlockerHelper.swift

```diff
@@ -73,7 +80,30 @@ class ContentBlockerHelper {
         return tab.isPrivate ? isEnabledInPrivateBrowsing : isEnabledInNormalBrowsing
     }
 
-    var isEnabledInNormalBrowsing: Bool {
+    var status: BlockerStatus {
+        guard isEnabled else {
+            return .Disabled
+        }
+        if stats.total == 0 {
+            guard let url = tab?.url else {
+                return .NoBlockedURLs
+            }
+            return isURLWhitelisted(url: url) ? .Whitelisted : .NoBlockedURLs
+        } else {
+            return .Blocking
+        }
+    }
+
+    var stats: TPPageStats = TPPageStats() {
+        didSet {
+            guard let tab = self.tab else { return }
+            if (stats.total == 0 && oldValue.total != 0) || stats.total == 1 {
+                TabEvent.post(.didChangeContentBlocking, for: tab)
```

Comment:
This is great, we need to land this and I'll rebase my patch on top of this code to use this event.